### ISSUE TITLE
Fine Tuning Embedding

### ISSUE THREAD

**Issue by yaskh** [UKPLab/sentence-transformers#353]
Hi,
Does this module provide the functionality to fine tune the pretrained model on my own data.
I want to fine tune the model on my own text data, then after fine tuning generate the embeddings which are more specific to my data. 
Any help is appreciated, thanks!

**Comment by farhaanbukhsh** [comment]
hey @yaskh  you can have look at https://github.com/UKPLab/sentence-transformers/issues/350

**Comment by yaskh** [comment]
Hi @farhaanbukhsh 
the issue you pointed to trains a mutli label classifier. My dataset doesn't contain labels. I there a way to train the embeddings on an unlabeled dataset?  
Thanks

**Comment by nreimers** [comment]
Hi @yaskh 
You need to be able to tell somehow if two sentences are similar or not. If you have purely sentences, without any labels or structure, a good  representation can sadly not be learned.

Best
Nils

**Comment by yucongo** [comment]
Hi @nreimers.

I am thinking of doing something similar as yaskh wanted to do, i.e., to fine-tune a model on my own text data. 

Would it help to fine-tune the model based on Masked Language Modeling?  It can at least help create embedding for out-of-vocabulary words?

Thanks a lot

**Comment by nreimers** [comment]
Hi @yucongo 
Do you have labeled data for your domain?

Yes MLM can help before you do fine-tuning with your data.

Have a look here:
https://github.com/UKPLab/sentence-transformers/tree/master/examples/unsupervised_learning

and in our paper:
https://arxiv.org/abs/2104.06979

where we tested strategies

**Comment by yucongo** [comment]
Thanks @nreimers. I'll take a look at  https://github.com/UKPLab/sentence-transformers/tree/master/examples/unsupervised_learning

No labeled data for the domain, just raw data.
